jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -1084,6 +1084,18 @@ public long getSessionTimeMax()
 
     public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingModes)
     {
+        if (sessionTrackingModes != null &&
+            sessionTrackingModes.size() > 1 &&
+            sessionTrackingModes.contains(SessionTrackingMode.SSL))
+        {
+            for (SessionTrackingMode sessionTrackingMode:sessionTrackingModes)
+            {
+                if (sessionTrackingMode != SessionTrackingMode.SSL)
+                {
+                    throw new IllegalArgumentException("sessionTrackingModes specifies a combination of SessionTrackingMode.SSL with a session tracking mode other than SessionTrackingMode.SSL");
+                }
+            }
+        }
```

Comment:
Isn't this more succinctly:

if (sessionTrackingModes != null 
    && sessionTrackingModes.size() > 1 
    && sessionTrackingModes.contains(SessionTrackingModes.SSL)
    throw new IllegalArgumentException ("sessionTrackingModes specifies a combination of SessionTrackingMode.SSL with a session tracking mode other than SessionTrackingMode.SSL");

In other words, if SessionTrackingMode.SSL is present, it must be the only member of the set?

Reply:
oh yes good catch as it's a Set we can have SSL only once :)